spec/context-loss.md

```diff
@@ -32,16 +32,20 @@ When the user agent detects that the backing storage associated with a Canvas2D
 3. Set *context lost* flag.
 4. Queue a task to perform the following steps:
     1. Fire a event `contextlost` at *canvas*
-    2. If the event's canceled flag is not set, abort these steps.
+    2. If the event's canceled flag is set, abort these steps. The backing buffer for the context will not be restored.
```

Comment:
what is the use case for not restoring the buffer?

Reply:
My understanding is that some applications are using extra canvases as distributed buffers for assembling large scenes. The use case here is that if the context fails (potentially due to GPU OOM) then the app will use the cancelation behavior to prevent the UA from "restarting" the "extra" canvases that may have contributed to the memory pressure in the first place (the app will dynamically scale down the number of extra distributed canvases in use in this scenario, possibly to zero).

In short, the flag optimizes the case where the canvas will no longer be used again, so the UA won't need to bother allocating and resetting it just so the app will then remove it and have it be GC'd later.

@macrabil does that sound right to you?

Reply:
Sounds right, yeah.

Reply:
Makes sense.